madmom/evaluation/key.py

```diff
@@ -0,0 +1,246 @@
+    14
+
+    """
+    tonic, mode = key_label.split()
+    key_class = _KEY_TO_SEMITONE[tonic.lower()]
+    if mode in ['minor', 'min']:
+        key_class += 12
+    elif mode in ['major', 'maj']:
+        key_class += 0
+    else:
+        raise ValueError('Unknown mode: {}'.format(mode))
+    return key_class
+
+
+def error_type(ann_key, det_key, strict_fifth=False):
+    """
+    Compute the evaluation score and error category for a predicted key
+    compared to the annotated key.
+
+    Categories and evaluation scores follow the evaluation strategy used
+    for MIREX (see http://music-ir.org/mirex/wiki/2017:Audio_Key_Detection).
+    There are two evaluation modes for the 'fifth' category: by default,
+    a detection falls into the 'fifth' category if it is the fifth of the
+    annotation, or the annotation is the fifth of the detection.
+    If `strict_fifth` is `True`, only the former case is considered. This is
+    the mode used for MIREX.
+
+    Parameters
+    ----------
+    ann_key : int
+        Annotated key class.
+    det_key : int
+        Detected key class.
+    strict_fifth: bool
+        Use strict interpretation of the 'fifth' category, as in MIREX.
+
+    Returns
+    -------
+    category, score : str, float
+        Error category and evaluation score.
+
+    """
+    ann_root = ann_key % 12
+    ann_mode = ann_key // 12
+    det_root = det_key % 12
+    det_mode = det_key // 12
+    major, minor = 0, 1
+
+    if det_root == ann_root and det_mode == ann_mode:
+        return 'correct', 1.0
+    if det_mode == ann_mode and ((det_root - ann_root) % 12 == 7):
+        return 'fifth', 0.5
+    if not strict_fifth and (det_mode == ann_mode and
+                             ((det_root - ann_root) % 12 == 5)):
+        return 'fifth', 0.5
+    if (ann_mode == major and det_mode != ann_mode and (
+            (det_root - ann_root) % 12 == 9)):
+        return 'relative', 0.3
+    if (ann_mode == minor and det_mode != ann_mode and (
+            (det_root - ann_root) % 12 == 3)):
+        return 'relative', 0.3
+    if det_mode != ann_mode and det_root == ann_root:
+        return 'parallel', 0.2
+    else:
+        return 'other', 0.0
+
+
+class KeyEvaluation(EvaluationMixin):
+    """
+    Provide the key evaluation score.
+
+    Parameters
+    ----------
+    detection : str
+        File containing detected key
+    annotation : str
+        File containing annotated key
+    strict_fifth : bool
+        Use strict interpretation of the 'fifth' category, as in MIREX.
+    name : str, optional
+        Name of the evaluation object (e.g., the name of the song).
+
+    """
+
+    METRIC_NAMES = [
+        ('score', 'Score'),
+        ('error_category', 'Error Category')
+    ]
+
+    def __init__(self, detection, annotation,
+                 strict_fifth=False, name=None, **kwargs):
+        self.name = name or ''
+        self.detection = key_label_to_class(open(detection).read().strip())
```

Comment:
It can't work with string key labels, because you cannot reliably differentiate between a file name string a key string (other than assuming that no-one calls a file "A major"...).

However, the function can handle `int` now.

`None` does not make sense in this context I think.

Reply:
That's true. It could be catched with a try/except clause, though. But I don't have any strong arguments for including it necessarily.

Reply:
Yes, but then the function would behave differently depending on the existence of a file in the current directory. After all, someone **might** call a file "a min"...

Reply:
Right. I just thought of any possibility that it behaves more like the other functions for loading annotations/detections. Maybe we can/should open the annotation/detection in files `bin/evaluate` and require open file handles in the loading functions...

Reply:
I added an issue for that:  #338